src/mongoose_rabbit_worker.erl

```diff
@@ -241,20 +176,9 @@ establish_rabbit_connection(Opts, Host) ->
             exit("connection to a Rabbit server failed")
     end.
 
--spec maybe_enable_confirms(Channel :: pid(), Opts :: proplists:proplist()) -> ok.
-maybe_enable_confirms(Channel, Opts) ->
-    case proplists:get_value(confirms_enabled, Opts) of
-        true ->
-            #'confirm.select_ok'{} =
-                amqp_channel:call(Channel, #'confirm.select'{}),
-            true;
-        false ->
-            false
-    end.
-
 -spec close_rabbit_connection(Connection :: pid(), Channel :: pid(),
                               Host :: jid:server()) ->
-    ok | term().
+    ok | no_return().
```

Comment:
This function should not have `no_return()`
It can either return `already_closed` or `ok` or `Error :: term()`  (from amq_connection:close() )

Reply:
When we try to close a connection that is already closed, according to [this]( https://github.com/rabbitmq/rabbitmq-erlang-client/blob/master/src/amqp_gen_connection.erl#L82), it ends up with an exception.